### PR TITLE
Feat/chfo/mergedylib6 runtime

### PR DESCRIPTION
**Pull request by cfogelklou**
This can supersede my previous pull request!

I pull-requested feat/chfo/mergedylib5 previously, but that wasn't rebased/tested against the latest robovm master.

Good luck with this!  This is the "runtime" branch.  I will shortly make a pull-request for the "compiler" branch.

Obviously it has taken us (me) a long time to release this, and in all likelyhood we might not be able to make too many changes to this due to the fact that it is almost summer!

### CODE REVIEW

**Comment by ntherning** [comment]
Thanks Chris! Will take a look and feed back to you next week.

File: vm/bc/src/MurmurHash3.c [hunk 1]
```diff
@@ -28,7 +28,17 @@
 
 #else	// defined(_MSC_VER)
 
-#define	FORCE_INLINE __attribute__((always_inline))
+// RoboVM note: Start change.
```
Comment (ntherning) [inline vm/bc/src/MurmurHash3.c]: Why is this needed on Linux? It's been working fine so far?

**Comment by ntherning** [comment]
I have merged this. Thanks!

File: vm/bc/src/MurmurHash3.c (hunk 1 above)
Comment (cfogelklou) [inline vm/bc/src/MurmurHash3.c]: I thought I had replied, but it looks like I didn't!  I just found out where to see this discussion - I had seen these comments in my "app" email but had to use google-fu to find them here.

This is needed because the code does not compile without -fPIC enabled on linux.  But I see you've merged this now, thanks!
